math/chinese_remainder_theorem.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file
+ * @brief C++ program to solve a system of linear congruences using the Chinese
+ * Remainder Theorem (CRT).
+ *
+ * @details
+ * The Chinese Remainder Theorem (CRT) is a mathematical theorem that provides a
+ * solution to a system of linear congruences. It allows us to find a unique
+ * solution to the system if the moduli are pairwise coprime (i.e., the greatest
+ * common divisor of any two moduli is 1).
+ *
+ * This implementation assumes that the moduli are pairwise coprime.
+ *
+ * @note For the system of congruences to have a solution, the moduli must be
+ * pairwise coprime. If they are not, the result obtained may not be valid.
+ *
+ * @author [Himesh](https://github.com/HimeshKohad)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+#include <vector>    /// for std::vector
+
+struct LinearCongruance {
+    const int a;
+    const int m;
```

Comment:
What happens if `a` or `m` are negative? What happens if `m` is zero?

Reply:
Thank you for your suggestion.
I think the code would work for the cases when a and m are negative because the modular arithmetic operator allows negative numbers to be represented in modulo. 
As for the case where m is zero, I have updated the code accordingly to handle that case.